base/array.jl

```diff
@@ -2,6 +2,38 @@
 
 ## array.jl: Dense arrays
 
+"""
+    Array{T}(dims)
+    Vector{T}(m)
+    Matrix{T}(m, n)
+
+`Array{T}(dims)` constructs an uninitialized dense array with element type `T`. `dims` may
+be a tuple or a series of integer arguments, each giving the size of the corresponding
+dimension. `Vector{T}(m)` and `Matrix{T}(m, n)` are stricly equivalent aliases to create one-
+and two-dimensional arrays.
+
+The syntax `Array(T, dims)` is also available, but deprecated.
+
+
```

Comment:
Defining a new docstring at the REPL would add it to `Main`'s docstring cache and then using `?` will concatenate docs from all modules, whereas defining extra ones in that file would add them to `Base`'s cache. Duplicates are only checked for within the current module. Maybe a more thorough check could be done, though that might get a bit expensive perhaps.

Perhaps the following would work:

```julia
" ... "
Array(::Tuple), Vector(::Integer), Matrix(::Integer, ::Integer)

" ... "
Array(itr), Vector(itr), Matrix(itr)
```

or something similar to that.

Appending to `convert`'s docs is probably a bad idea :)